src/haddock/libs/libhpc.py

```diff
@@ -127,6 +133,24 @@ def cancel(
             cmd = f"scancel {self.job_id}"
             _ = subprocess.run(shlex.split(cmd), capture_output=True)
 
+    def restart(self):
+        """Restart an execution."""
+        # Cancel previous job
+        self.cancel()
+        # Increase number of tries for this job
+        self.tries += 1
+        # Check if must too many tries
+        if self.tries > MAX_JOB_TRIES:
+            self.job_status = "failed"  # Set failed flag
+            for fpath in [
+                self.job_fname,
+                self.job_fname.with_suffix('.out'),
+                self.job_fname.with_suffix('.err'),
+                    ]:
+                fpath.unlink(missing_ok=True)
```

Comment:
These files are in the `#SBATCH` header, slurm will take care of them, this part is not needed and it is assuming the suffixes of the files, this `missing_ok` should not be there.

What I meant for files that are linked to the job could be for example the PDB files that are generated by a specific execution - or whatever other file. So you can either parse the job file and find what is the expected output or add it as an attribute to the `CNSJob` class and handle it here.

Reply:
ok , has been removed